File: Lab04/lab04.py
###################################################
# MC102 - Algoritmos e Programação de Computadores
# Laboratório 4 - Controle de Estoque
# Nome: Matheus Alves de Andrade
# RA: 256296
###################################################

# leitura da sequência de compras e vendas
estoque = 0
vendas = 0

while True:
    x = int(input(""))
    if x > 0:
            estoque = estoque + x
            continue
    elif x < 0:
        if estoque - abs(x) < 0:
            print("Quantidade indisponível para a venda de " + str(abs(x)) + " unidades.")
            continue
        else:
            estoque = estoque - abs(x)
            vendas += 1
            continue
    else:
            print("Quantidade de vendas realizadas: " + str(vendas))
            print("Quantidade em estoque: " + str(estoque))
            break
        

File: Lab04/lab04.f90
! ###################################################
! # MC102 - Algoritmos e Programação de Computadores
! # Laboratório 4 - Controle de Estoque
! # Nome: Matheus Alves de Andrade
! # RA: 256296
! ###################################################

program INVMAN
    ! Program INVentory MANager

    ! Manages the inventory for a single product by reading a 
    ! series of purchases and sales of the product, outputting
    ! the amount of sales and remaining inventory. Sales are
    ! entered as a negative integer, and purchases are entered as
    ! a positive integer.


    ! Declarations
    integer INVTRY, SALES, INPUT ! Inventory and Sales
    INVTRY = 0 ! Initial inventory of 0
    SALES = 0 ! Initial sales value of 0

    do
        read *, INPUT

        if ( INPUT .gt. 0) then
            INVTRY = INVTRY + INPUT
            continue
        else if ( INPUT .lt. 0) then
            if ( INVTRY - ABS(INPUT) .lt. 0) then
                print *, "Quantidade indisponível para a venda de ", ABS(INPUT), " unidades."
                continue
            else
                INVTRY = INVTRY - ABS(INPUT)
                SALES = SALES + 1
                continue                
            end if
        else
            print *, "Quantidade de vendas realizadas: ", SALES
            print *, "Quantidade em estoque: ", INVTRY
            stop
        end if
    end do
    
end program INVMAN